lib/src/db.rs

```diff
@@ -628,6 +637,179 @@ impl Storelike for Db {
             }
         }
     }
+
+    fn add_resource(&self, resource: &Resource) -> AtomicResult<()> {
```

Comment:
I'm a bit of confused with the next few hundred LOC. Seems like this is the same as the functions in the `StoreLike` Trait. Why have you copied them? Did it not compile if you kept them in the Trait?

Reply:
That's interesting because I had this question for you - I noticed the same methods in StoreLike and db.rs. Did I grab the wrong branch to play with? I believe db.rs only triggered with --all-features flag.